build_daemon/lib/constants.dart

```diff
@@ -24,6 +24,11 @@ String daemonWorkspace(String workingDirectory) {
   return p.joinAll(segments);
 }
 
+/// The number of times to retry reading the various state files.
+///
+/// This prevents the likelihood of race conditions.
+const maxRetries = 50;
```

Comment:
also seems like this probably shouldn't be a public thing?

Reply:
Yeah it really shouldn't but I didn't know where else we should shove this so I thought it was pretty harmless. I'll see where it ends up with the suggested refactors.